### ISSUE TITLE
Bulk operations do not work for Batches

### ISSUE THREAD

**Issue by jpcamara** [bensheldon/good_job#1255]
I don't _think_ this is intentional behavior. But take the following batch code:

```rb
class GoodJobJob < ApplicationJob
  def perform(arg)
    puts "Hi #{arg}!"
  end
end
```

```rb
class GoodBatchCallbackJob < ApplicationJob
  def perform(batch, params)
    puts "Here's how many jobs we ran: #{batch.active_jobs.size}"
  end
end
```

```rb
GoodJob::Batch.enqueue(GoodJobJob.new("0"), on_finish: GoodBatchCallbackJob) do
  GoodJob::Bulk.enqueue([
    GoodJobJob.new("1"),
    GoodJobJob.new("2"),
    GoodJobJob.new("3")
  ])

  GoodJob::Adapter.new.enqueue_all([
    GoodJobJob.new("4"),
    GoodJobJob.new("5"),
    GoodJobJob.new("6")
  ])

  ActiveJob.perform_all_later([
    GoodJobJob.new("7"),
    GoodJobJob.new("8"),
    GoodJobJob.new("9")
  ])

  GoodJobJob.perform_later("10")
  GoodJob::Adapter.new.enqueue(GoodJobJob.new("11"))
end
```

Above we have a job `GoodJobJob`, which is being run as part of a batch. It's invoked in several ways:

- Using the `Batch.enqueue` `active_jobs` argument ✅ 
- Using `GoodJob::Bulk.enqueue` ❌ 
- Using `GoodJob::Adapter.new.enqueue_all` ❌ 
- Using `ActiveJob.perform_all_later` ❌ 
- Using regular `perform_later` ✅ 
- Using `GoodJob::Adapter.new.enqueue` ✅ 

Running this code, we get the following output:

```
Hi 2!
Hi 1!
Hi 4!
Hi 3!
Hi 5!
Hi 6!
Hi 7!
Hi 8!
Hi 9!
Hi 0!
Hi 11!
Hi 10!
Here's how many jobs we ran: 3
```

The only three jobs that are recognized by the batch are any jobs performed without bulk operations. `perform_all_later`, `enqueue_all` and `Bulk.enqueue` (which I know all effectively use the same `Bulk.enqueue` code) get ignored.

From my investigation, it looks like the only place the batch id gets utilized is inside of `GoodJob::Execution.enqueue`, in its `enqueue_args` method. 

I'm assuming this is an oversight, and not intentional behavior? Thanks!

**Comment by jpcamara** [comment]
Ah, I see. I looked for issues but didn't think to check the discussions as well sorry! I could probably add a doc about it if that helps/is desired. Raising could be nice as well.

For my particular use-case I was not seeing jobs get enqueued and went around in circles for a bit until I realized `perform_all_later` was the issue. I only tried the other options to test the overall behavior.

For me it wasn't for optimization - I had serialized some active jobs for later execution and when I deserialized I was using `perform_all_later` to run the instances. I've found other ways around it now. I'll close the issue if it's working as intended.
